src/app/core/admin/admin-entity-details/admin-entity-field/admin-entity-field.component.ts

```diff
@@ -281,6 +283,14 @@ export class AdminEntityFieldComponent implements OnInit {
 
     const fieldId = this.fieldIdForm.getRawValue();
 
+    // If a default value is configured, convert it to the database format
+    const defaultValueConfig = this.entitySchemaField?.defaultValue?.config;
+    if (defaultValueConfig?.value) {
+      defaultValueConfig.value = this.entitySchemaService.valueToDatabaseFormat(
+        defaultValueConfig.value,
+        this.entitySchemaField,
+      );
+    }
```

Comment:
why does this have to happen here? (I would see such a defaultValue specific task as the responsibility for the admin-default-value component)

And shouldn't the defaultValue stored in the entity schema by definition already be in database-format anyway?

Reply:
Sure, I will move this to the admin-default-value component

While trying to save the updated field schema(defaultvalueconfig), right now it's not already in database format, that's why I am doing this -
![image](https://github.com/user-attachments/assets/c28ae94e-8208-4190-8b11-e1e299c73e74)

Reply:
oh, I see. Are we saving it in this format to the schema? Or do we transform it somewhere else along the way?

I thing it would be best to keep this as an internal implementation detail of admin-default-static and just expose the database-format to the outside

Reply:
I think we have been saving this format to the schema as, is until now (probably because it only supported simple text, so there was no need for transformation).
But now I’ve added the logic inside the default-value-component itself to transform the value before emitting/saving it.